### DISCUSSION TITLE
Questions about setting up a testing environment.

### DISCUSSION THREAD

**Discussion by NJRoadfan** [OpenPrinting/libcups#73] [Q&A] [answered]
Hello,

I am [working on](https://github.com/Netatalk/netatalk/issues/663) updating some code to support libcups3. I am at the point that I would like to test against the actual libcups3 library. How would one go about doing this? Can libcups3 co-exist with an existing CUPS 2.x install? Or do I have to start fresh with a new Linux VM without any CUPS packages installed and build libcups3?

Also, is the cups-local server needed to print? If so, I'll have to build/install it as well. My existing code calls libcups to submit a job (ultimately a PDF) via cupsCreateDestJob() and cupsStartDestDocument().

**Comment by michaelrsweet** [comment, marked as answer]
Hi, the default build of libcups 3.x will install so that it can co-exist with CUPS 2.x. Use "pkg-config --something cups" to build against CUPS 2.x and "pkg-config --something cups3" to build against libcups 3.x.

WRT cups-local, it isn't needed for printing with libcups, but (eventually) if cups-local is running then libcups will default to using it instead of cupsd.

**Comment by NJRoadfan** [comment]
No problems compiling and installing. Right now I'm trying to figure out why I get a segfault when running the simple example program (definitely linking to libcups3) which only calls cupsEnumDests(). System is running CUPS 2.3.3op2.

EDIT: Switched from master to 3.0b2. No more segfault, but program hangs after listing printers.

**Reply by michaelrsweet** [reply]
Hmm, I just reproduced this on macOS (CUPS 2.3.4). Seems to be an issue saving the printer-state-change-time value - can you file an issue and I'll see about fixing this? Thanks!

**Reply by michaelrsweet** [reply]
Nevermind, Github has a "create issue from discussion" action, so I've created issue #74 to track this.
